src/js/actions/libraries.js

```diff
@@ -67,42 +69,51 @@ define(function (require, exports) {
         var currentLayer = currentLayers.first(),
             IMAGE_ELEMENT_TYPE = "application/vnd.adobe.element.image+dcx",
             REPRESENTATION_TYPE = "image/vnd.adobe.photoshop";
-
-        currentLibrary.beginOperation();
-
-        var newElement = currentLibrary.createElement(currentLayer.name, IMAGE_ELEMENT_TYPE),
-            representation = newElement.createRepresentation(REPRESENTATION_TYPE, "primary"),
-            previewSize = {
-                w: 248,
-                h: 188
-            };
-
-        // FIXME: Mac/Win temporary locations!
-        var exportObj = libraryAdapter.exportLayer("/tmp/", "/tmp/preview.png",
-            currentLayer.name, previewSize);
-
-        return descriptor.playObject(exportObj)
-            .bind(this)
+        
+        var newElement;
+
+        return os.getTempFilename()
```

Comment:
You definitely want to use this API with your own filename (otherwise it'll return `tempfolder/1`). So add in preview.png here, and then use a utility function in path (or write one if it doesn't exist) to extract basepath.